adam-cli/src/main/scala/org/bdgenomics/adam/cli/TransformGenotypes.scala

```diff
@@ -0,0 +1,132 @@
+  var forceShuffle: Boolean = false
+
+  @Args4jOption(required = false, name = "-sort_on_save", usage = "Sort VCF output by contig index.")
+  var sort: Boolean = false
+
+  @Args4jOption(required = false, name = "-sort_lexicographically_on_save", usage = "Sort VCF output by lexicographic order. Conflicts with -sort_on_save.")
+  var sortLexicographically: Boolean = false
+
+  @Args4jOption(required = false, name = "-single", usage = "Save as a single VCF file.")
+  var asSingleFile: Boolean = false
+
+  @Args4jOption(required = false, name = "-defer_merging", usage = "Defers merging single file output.")
+  var deferMerging: Boolean = false
+
+  @Args4jOption(required = false, name = "-disable_fast_concat", usage = "Disables the parallel file concatenation engine.")
+  var disableFastConcat: Boolean = false
+
+  @Args4jOption(required = false, name = "-stringency", usage = "Stringency level for various checks; can be SILENT, LENIENT, or STRICT. Defaults to STRICT.")
+  var stringency: String = "STRICT"
+
+  // must be defined due to ADAMSaveAnyArgs, but unused here
+  var sortFastqOutput: Boolean = false
+}
+
+/**
+ * Convert a file with genotypes into corresponding ADAM format and vice versa.
+ */
+class TransformGenotypes(val args: TransformGenotypesArgs)
+    extends BDGSparkCommand[TransformGenotypesArgs] {
+  val companion = TransformGenotypes
+  val stringency = ValidationStringency.valueOf(args.stringency)
+
+  /**
+   * Coalesce the specified GenomicRDD if requested.
+   *
+   * @param rdd GenomicRDD to coalesce.
+   * @return The specified GenomicRDD coalesced if requested.
+   */
+  private def maybeCoalesce[U <: GenomicRDD[_, U]](rdd: U): U = {
+    if (args.coalesce != -1) {
+      log.info("Coalescing the number of partitions to '%d'".format(args.coalesce))
+      if (args.coalesce > rdd.rdd.partitions.length || args.forceShuffle) {
+        rdd.transform(_.coalesce(args.coalesce, shuffle = true))
+      } else {
+        rdd.transform(_.coalesce(args.coalesce, shuffle = false))
+      }
+    } else {
+      rdd
+    }
+  }
+
+  /**
+   * Sort the specified GenomicRDD if requested.
+   *
+   * @param rdd GenomicRDD to sort.
+   * @return The specified GenomicRDD sorted if requested.
+   */
+  private def maybeSort[U <: GenomicRDD[_, U]](rdd: U): U = {
+    if (args.sort) {
+      log.info("Sorting before saving")
+      rdd.sort()
+    } else if (args.sortLexicographically) {
+      log.info("Sorting lexicographically before saving")
+      rdd.sortLexicographically()
+    } else {
+      rdd
+    }
+  }
+
+  def run(sc: SparkContext) {
+    require(!(args.sort && args.sortLexicographically),
+      "Cannot set both -sort_on_save and -sort_lexicographically_on_save.")
+
+    val genotypes = sc.loadGenotypes(
+      args.inputPath,
+      optPredicate = None,
+      optProjection = None,
+      stringency = stringency)
+
+    if (args.outputPath.endsWith(".vcf")) {
```

Comment:
There wasn't a check for compressed formats before, and I don't think we support writing compressed VCF through HadoopBAM.  With a code change to use `FileExtensions.isVcfExt`
```bash
$ ./bin/adam-submit transformVariants \
  -single \
  adam-core/src/test/resources/small.vcf \
  small.vcf.gz

$ gzcat small.vcf.gz 
gzcat: small.vcf.gz: not in gzip format

$ head small.vcf.gz 
##fileformat=VCFv4.2
##FILTER=<ID=IndelFS,Description="FS > 200.0">
##FILTER=<ID=IndelQD,Description="QD < 2.0">
...
```
Do we have to set a configuration parameter or otherwise kick HadoopBAM to write .gz or .bgz/.bgzf?